api/post.go

```diff
@@ -384,11 +384,19 @@ func handleWebhookEvents(c *Context, post *model.Post, team *model.Team, channel
 	relevantHooks := []*model.OutgoingWebhook{}
 	for _, hook := range hooks {
 		if hook.ChannelId == post.ChannelId {
-			if len(hook.TriggerWords) == 0 || hook.HasTriggerWord(firstWord) {
+			if len(hook.TriggerWords) == 0 {
+				relevantHooks = append(relevantHooks, hook)
+			} else if hook.TriggerWhen == 0 && hook.HasTriggerWord(firstWord) {
```

Comment:
Can we get some consents for these values? Maybe have a function that encapsulates this logic? This whole section is really hard to understand.